Comments: Prompt to update Jetpack to edit comments on JP<5.3 sites

Copons opened the pull request:
On Jetpack <5.3 sites we can't easily retrieve the comment's `raw_content`.
Instead of flooding the `CommentDetail` component with conditionals and introducing a new Query component for old Jetpack versions, we simply display an update nudge.

See: p1HpG7-4aq-p2

<img width="670" alt="screen shot 2017-09-06 at 18 34 36" src="https://user-images.githubusercontent.com/2070010/30125962-320dc222-9332-11e7-9e03-ab717caaea55.png">

cc @Automattic/lannister

dmsnell commented:
@copons make sure you rebase this against **master** now that the tests are fixed and then wait for the green-light before merging ✔️

rodrigoi commented on client/blocks/comment-detail/comment-detail-edit.jsx:
```diff
@@ -85,13 +94,29 @@ export class CommentDetailEdit extends Component {
+						) }
+					>
+						<NoticeAction href={ `/plugins/jetpack/${ siteSlug }` }>
+							{ translate( 'Update Now' ) }
```
I think it'll be cool to update Jetpack in place using `PluginActions` like: https://github.com/Automattic/wp-calypso/blob/78ee303a91bf5fe571c315bfb552c5b946144b66/client/my-sites/plugins/plugin-meta/index.jsx#L389

It's a lot more complicated (maybe for **M3**), but considering number of actions the user has to take to reach this notice, it may be nice to prevent it from leaving the page.
Not a blocker, but something to think about.

Copons replied:
I wasn't aware of such thing!
Let's open an issue on M2, and if it's _that_ complicated we can move it to M3, since it's not a blocker whatsoever. 🙂

Copons commented:
> make sure you rebase this against master now that the tests are fixed and then wait for the green-light before merging ✔️

@dmsnell I was sure I opened this branch right after pulling that fix! Maybe I was quicker than I thought, and opened before it was merged... 🤔

kwight commented:
Are these sorts of notices ever nested like this? I think of them as more "global" in display.

Copons commented:
> Are these sorts of notices ever nested like this? I think of them as more "global" in display.

I thought about this (as your mock used a global notice), but I believe it would be lost in the UI when on big screens.
Having a notice localized to the very feature it refers to, makes it more obvious imho.

But, it's a good point nonetheless.
Looping in design people for an opinion!
cc @folletto

folletto commented:
> Are these sorts of notices ever nested like this? I think of them as more "global" in display.

Yes.

This is not a notice kind of message, this is a state message. 
It should be part of the card design.

There are different appoaches I can see here:

1. We remove Edit button entirely.
2. We keep edit, but it's disabled and has a tooltip text saying "Upgrade Jetpack to edit"
3. We keep edit, it still opens, but instead of the form we show the message and the upgrade link. Just text on white. Maybe we can add an illustration if we feel it's too light.
4. We keep edit, it still opens, it still show a disabled form, same message as above, but instead of a notice it's simply icon + text in `$alert-red`, and the link is simply a underlined `a`.

kwight commented:
Ooooh, I vote for 3 or 4.

Copons commented:
I vote 4 actually!
I like how the disabled form functions as a sort of "mouth-waterer".
You have the feature ready, you can see it, you can almost use it, but... nnnnnope! You have to update JP first!

Copons commented:
Updated the PR: removed the Notice and added a simple red text as suggested by @folletto.

<img width="670" alt="screen shot 2017-09-06 at 18 34 36" src="https://user-images.githubusercontent.com/2070010/30125962-320dc222-9332-11e7-9e03-ab717caaea55.png">

gwwar commented on client/blocks/comment-detail/style.scss:
```diff
@@ -586,6 +586,18 @@ a.comment-detail__author-more-element {
+		margin-left: 8px;
+		margin-right: 8px;
+
+		a {
```
If we give the link a classname, we can break this out of the nesting. We only expect one link in the notice, correct?

gwwar commented on client/blocks/comment-detail/comment-detail-edit.jsx:
```diff
@@ -85,13 +93,25 @@ export class CommentDetailEdit extends Component {
 				<div className="comment-detail__edit-buttons">
 					<FormButton
 						compact
+						disabled={ ! isEditCommentSupported }
```
Should we also check for this in the Author fields?

Copons replied:
Damn, good catch!
I was testing on a comment I was sure was from an unregistered user, but it wasn't and I was tricked by the fact that the author fields were already disabled.

folletto commented:
Looks good 👍
